holmes/plugins/toolsets/mcp/toolset_mcp.py

```
@@ -92,6 +112,363 @@ class MCPMode(str, Enum):
+                self.end_headers()
+                self.wfile.write(b"<h1>Authenticated! You can close this tab.</h1>")
+            else:
+                result["error"] = params.get("error", ["unknown"])[0]
+                result["error_description"] = params.get("error_description", [""])[0]
+                self.send_response(400)
+                self.send_header("Content-Type", "text/html")
+                self.end_headers()
+                self.wfile.write(f"<h1>Error: {result['error']}</h1>".encode())
+            callback_event.set()
+
+        def log_message(self, format, *args):
```

Ruff (0.15.9)

[error] Function argument `format` is shadowing a Python builtin

(A002)